sea-orm-migration/src/migrator.rs

```diff
@@ -289,6 +289,70 @@ pub trait MigratorTrait: Send {
 
         Ok(())
     }
+
+    /// Apply or Rollback migrations to version
+    async fn change_to_version(db: &DbConn, version: &str) -> Result<(), DbErr> {
+        Self::up_to_version(db, version).await?;
+        Self::down_to_version(db, version).await
+    }
+
+    /// Apply migrations to version
+    async fn up_to_version(db: &DbConn, version: &str) -> Result<(), DbErr> {
+        let steps = Self::get_steps(db, MigrationStatus::Pending, version).await?;
+        if steps > 0 {
+            Self::up(db, Some(steps)).await
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Rollback migrations to version
+    async fn down_to_version(db: &DbConn, version: &str) -> Result<(), DbErr> {
+        let steps = Self::get_steps(db, MigrationStatus::Applied, version).await?;
+        if steps > 1 {
+            Self::down(db, Some(steps - 1)).await
+        } else {
+            Ok(())
+        }
+    }
+
+    async fn get_steps(db: &DbConn, status: MigrationStatus, version: &str) -> Result<u32, DbErr> {
+        let mut index = 0;
+        let mut matched = false;
+        let migrations = Self::get_migration_with_status(db).await?;
+        match status {
+            MigrationStatus::Pending => {
+                for migration in migrations {
+                    if migration.status == MigrationStatus::Pending {
+                        index += 1;
+                    }
+                    if migration.migration.name() == version {
+                        matched = true;
+                        break;
+                    }
+                }
+            }
+            MigrationStatus::Applied => {
+                for migration in migrations.into_iter().rev() {
+                    if migration.status == MigrationStatus::Applied {
+                        index += 1;
+                    }
+                    if migration.migration.name() == version {
+                        if migration.status == MigrationStatus::Pending {
+                            index += 1;
+                        }
+                        matched = true;
+                        break;
+                    }
+                }
+            }
```

Comment:
Same here. Using `get_applied_migrations` to get the list of applied migrations. I think the logic where `migration.status == MigrationStatus::Pending` still increment the `index` by 1 is faulty. As the `down` method is counting on applied migration only and all pending migrations are ignored.

Consider the case where we have three migrations:
1. Migration 1: Applied
2. Migration 2: Pending
3. Migration 3: Applied
We should rollback 2 migrations instead of 3

Reply:
Down to Migration 1, after that Migration 1 is Applied or Pending,   In practical application, I think Applied is more reasonable，if Pending,  I don't know what the current status is，What about your suggestion？